client/src/App.jsx

```diff
@@ -0,0 +1,37 @@
+import { useState } from "react";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import Homepage from "./pages/Homepage.jsx";
+import SinglePost from "./pages/SinglePost.jsx";
+import Write from "./pages/Write.jsx";
+import DashboardLayout from "./components/DashboardLayout.jsx";
+
+const router = createBrowserRouter([
```

Comment:
What a great opportunity for me to learn about React Router!

Reply:
:) agreed!